app/javascript/components/SignUp.jsx

```diff
@@ -30,28 +32,39 @@ class SignUp extends Component {
     };
 
     async formSubmit(formData){
-        var data = new FormData(formData);
+        console.log(this.state)
+        var data = this.state;
         await fetch('/signup_user', {
-            Accept: 'application/json',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
             method: "POST",
-            mode: "cors",
-            body: data
-        }).then(response => response.text())
-        .then(res => console.log(res))
+            body: JSON.stringify(data)
+        }).then(response => response.json())
+          .then(() => <Tweetes />);
     }
 
     render() {
         return(
             <form onSubmit={ this.handleSubmit }>
-                <input type="text" name="name" value={ this.state.name } placeholder="Name"
+                <label htmlFor="name">Name: </label>
```

Comment:
nit: I see a lot of duplication here. We could reduce the duplicates by creating a reusable component for inputs. For example: 

```
function Input({ label, htmlFor, inputID, name, value, onChange, placeholder, type, required }) {
   return (
      <label htmlFor={ htmlFor }>{ label }</label>
      <input type={ type } id={ inputID } name={ name } value={ value } onChange={onChange} placeholder={ placeholder } required={ required } />
   )
}
```